packages/runtime-core/src/component.ts

```diff
@@ -988,9 +988,12 @@ export function formatComponentName(
 ): string {
   let name = getComponentName(Component)
   if (!name && Component.__file) {
-    const match = Component.__file.match(/([^/\\]+)\.\w+$/)
-    if (match) {
-      name = match[1]
+    const match = Component.__file.match(/(\/(\w+))?(\/(\w+))\.\w+$/)
+    if(match?.at(4) === "Index") {
```

Comment:
[Array.prototype.at](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at) is available in ES2022. Please use `match?.[idx]` instead.

Reply:
I think it should have the same logic for lowercase `index`.

Reply:
Yes, You are right!